### PR TITLE
Uploading/editing dats file

### PR DESCRIPTION
**Pull request by SanjeevLakhwani**
Now one can upload and edit dats-file which is also checked whether the file is a proper json

### CODE REVIEW

**Comment by davidlougheed** [comment]
the DATS file is stored as stringified JSON in katsu right now unfortunately, yeah
it really should not be... but that's a katsu change

File: src/components/datasets/DatasetForm.js [hunk 1]
```diff
@@ -9,26 +9,47 @@ import DataUseInput from "../DataUseInput";
...
+    }
+};
+
 const DatasetForm = ({ style, initialValue, form }) => {
     return (
         <Form style={style || {}}>
             <Item label="Title">
                 {form.getFieldDecorator("title", {
-                    initialValue: (initialValue || { title: "" }).title || "",
+
```
Comment (davidlougheed) [inline src/components/datasets/DatasetForm.js:26]: needless whitespace changes

File: src/components/datasets/DatasetForm.js [hunk 2]
```diff
@@ -9,26 +9,47 @@ import DataUseInput from "../DataUseInput";
...
+                    initialValue:
+                        (initialValue || { description: "" }).description || "",
+                        
                     rules: [{ required: true }],
                 })(<Input.TextArea placeholder="This is a dataset" />)}
             </Item>
             <Item label="Contact Information">
                 {form.getFieldDecorator("contact_info", {
-                    initialValue: (initialValue || { contact_info: "" }).contact_info || "",
+                 initialValue: (initialValue || { contact_info: "" }).contact_info || "",
```
Comment (davidlougheed) [inline src/components/datasets/DatasetForm.js:43]: whitespace

File: src/components/datasets/DatasetForm.js [hunk 3]
```diff
@@ -9,26 +9,47 @@ import DataUseInput from "../DataUseInput";
...
+
+                    initialValue: initialValue?.title || "",
+
                     rules: [{ required: true }, { min: 3 }],
                 })(<Input placeholder="My Dataset" size="large" />)}
             </Item>
             <Item label="Description">
                 {form.getFieldDecorator("description", {
-                    initialValue: (initialValue || { description: "" }).description || "",
+
```
Comment (davidlougheed) [inline src/components/datasets/DatasetForm.js:34]: whitespace

File: src/components/datasets/DatasetForm.js [hunk 4]
```diff
@@ -9,26 +9,47 @@ import DataUseInput from "../DataUseInput";
...
+                    initialValue:
+                        (initialValue || { description: "" }).description || "",
```
Comment (davidlougheed) [inline src/components/datasets/DatasetForm.js:36]: if youre changing this line, use null coalescing here like with title

File: src/components/datasets/DatasetForm.js [hunk 5]
```diff
@@ -9,26 +9,47 @@ import DataUseInput from "../DataUseInput";
...
-                    initialValue: (initialValue || { contact_info: "" }).contact_info || "",
+                 initialValue: (initialValue || { contact_info: "" }).contact_info || "",
                 })(<Input.TextArea placeholder={"Name\nInfo@c3g.ca"} />)}
             </Item>
+            <Item label="DATS File">
+                {form.getFieldDecorator("dats_file", {
+                    initialValue: initialValue?.dats_file || "",
+                    rules: [{ validator: validateJson }, { min: 2}],
+                })(<Input.TextArea />)}
+
```
Comment (davidlougheed) [inline src/components/datasets/DatasetForm.js:51]: whitespace

**Review by davidlougheed — CHANGES_REQUESTED** [review]

**Review by gsfk — APPROVED** [review]

**Pull request merged**